Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -180,5 +219,13 @@ private class VerificationProgressReporter : IVerificationProgressReporter {
         publisher.SendStatusNotification(document, CompilationStatus.VerificationStarted, message);
       }
     }
+
+    ImplementationId GetImplementationId(Implementation implementation) {
+      var prefix = implementation.Name.Split(Translator.NameSeparator)[0];
```

Comment:
Good question. I think the Name would serve as a reasonable Id.

But what if you rename a module? If you use the name as an Id, then you lose the "Stale state" for everything in that module, but if you use position, then you maintain it. Also, if you rename a method, you can keep the "stale state" for it.

I should add some tests for this though ^^

Reply:
I understand better some implications thanks to your explanation now.
In that case, do we really need the ID? Can't we just keep the position to be the ID? That way, any renaming will still make it possible to migrate diagnostics.

Reply:
The `ImplementationId` serves to distinguish between different implementations under the same verifiable declaration, since those implementation have the same position. The string part of the Id only contains strings like "Impl" or "WellFormed"